### PR TITLE
Searching new, optional, wish list notes

### PR DESCRIPTION
**Pull request by 48klocs**
There's a new, optional, line ending: `#notes:value`, where `value` can be, well, whatever value the person wants to tappity tap on in.

This addresses issue #4306 - the rough idea is that if people want to know that a given roll is getting the wish list thumbs up because they thought it'd be good for PvP or raids or whatever, then they can add notes to the rolls in the source file they load and this will match on up as they're found.

Spoiler alert: newer translations I do of panda's/merc's work will probably have PvP/PvE notes on them.

### CODE REVIEW

**Comment by 48klocs** [comment]
Thanks for beating me to the punch on the fix!

**Comment by bhollis** [comment]
Got a screenshot of how this ends up working?

File: src/app/wishlists/types.ts [hunk 1]
```diff
@@ -27,6 +27,9 @@ export interface CuratedRoll {
    * sure that at least every perk asked for is there.
    */
   isExpertMode: boolean;
+
+  /** Optional notes from the curator. */
+  notes: string | undefined;
```
Comment (bhollis) [inline src/app/wishlists/types.ts]: This can be written `notes?: string`.

**Comment by 48klocs** [comment]
I realized I forgot to surface notes after. Easy enough.

**Comment by 48klocs** [comment]
wish list notes

Touched it up to surface wish list notes in the item popup (searching on invisible data isn't the greatest).

**Pull request merged**